File: configlibrary/src/main/java/com/shmedo/configlibrary/iot/cmd/IOTCommandResult.java
package com.shmedo.configlibrary.iot.cmd;

import com.shmedo.configlibrary.iot.enums.IOTCommandType;

/**
 * 创建者:   gonghe <br/>
 * 创建时间:  2020/8/31 <br/>
 * 描述：     TODO
 */
public class IOTCommandResult<T> {
    public static final String COMMAND_HEADER = "$cmd=";
    public static final int RESULT_MIN_LENGTH = 5;
    public static final String ERROR_FLAG = "result=fail";

    /**
     * 命令执行是否成功
     */
    private boolean success;
    /**
     * 执行失败的错误信息
     */
    private String message;

    /**
     * 命令类型
     */
    private IOTCommandType commandType;
    /**
     * 解析结果数据
     */
    private T result;

    public boolean isSuccess() {
        return success;
    }

    public void setSuccess(boolean success) {
        this.success = success;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public IOTCommandType getCommandType() {
        return commandType;
    }

    public void setCommandType(IOTCommandType commandType) {
        this.commandType = commandType;
    }

    public T getResult() {
        return result;
    }

    public void setResult(T result) {
        this.result = result;
    }
}

File: lib_cmd/src/main/java/com/shmedo/lib/cmd/base/iot_cmd/parser/IOTCommandResult.kt
package com.shmedo.lib.cmd.base.iot_cmd.parser

import com.shmedo.lib.cmd.base.iot_cmd.enums.IOTCommandType

/**
 * 创建者:   gonghe <br/>
 * 创建时间:  2023/9/22 <br/>
 * 描述：     物联网指令解析后结果
 */

/**
 * IOT 指令执行结果密封类
 * @param T 结果数据类型
 */
sealed class IOTCommandResult<out T> {
    /**
     * 指令执行成功
     * @property data 结果数据
     * @property commandType 指令类型
     */
    data class Success<out T>(
        val data: T,
        val commandType: IOTCommandType
    ) : IOTCommandResult<T>()

    /**
     * 指令执行失败
     * @property message 错误信息
     * @property commandType 指令类型,可能为空
     */
    data class Failure(
        val message: String,
        val commandType: IOTCommandType? = null
    ) : IOTCommandResult<Nothing>()
}
